pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2.go

```diff
@@ -142,21 +177,239 @@ func (pq *AnalysisPriorityQueueV2) run() {
 		case <-pq.ctx.Done():
 			statslogutil.StatsLogger().Info("Priority queue stopped")
 			return
+		case <-dmlChangesFetchInterval.C:
+			statslogutil.StatsLogger().Info("Start to fetch DML changes of jobs")
+			pq.ProcessDMLChanges()
 		case <-timeRefreshInterval.C:
 			statslogutil.StatsLogger().Info("Start to refresh last analysis durations of jobs")
 			pq.RefreshLastAnalysisDuration()
 		}
 	}
 }
 
+// ProcessDMLChanges processes DML changes.
+// Note: This function is thread-safe.
+func (pq *AnalysisPriorityQueueV2) ProcessDMLChanges() {
+	pq.mu.Lock()
+	defer pq.mu.Unlock()
+
+	if err := statsutil.CallWithSCtx(pq.statsHandle.SPool(), func(sctx sessionctx.Context) error {
+		parameters := exec.GetAutoAnalyzeParameters(sctx)
+		if err := pq.setAutoAnalysisTimeWindow(parameters); err != nil {
+			return errors.Trace(err)
+		}
+		if !pq.isWithinTimeWindowWithoutLock() {
+			statslogutil.StatsLogger().Debug("Not within the auto analyze time window, skip processing DML changes")
+			return nil
+		}
+
+		start := time.Now()
+		defer func() {
+			statslogutil.StatsLogger().Info("DML changes processed", zap.Duration("duration", time.Since(start)))
+		}()
+
+		// We need to fetch the next check version with offset before fetching new DML changes.
+		// Otherwise, we may miss some DML changes happened during the process.
+		newMaxVersion := pq.statsHandle.GetNextCheckVersionWithOffset()
+		values := pq.statsHandle.Values()
+		lastFetchTimestamp := pq.lastDMLUpdateFetchTimestamp
+		for _, value := range values {
+			if value.Version > lastFetchTimestamp {
```

Comment:
Why do we need this condition? Could you add some comments.

Reply:
Added.